app/controllers/admin/accounts_controller.rb

```diff
@@ -0,0 +1,46 @@
+module Admin
+  class AccountsController < BaseController
+    load_and_authorize_resource
+
+    def index
+      @q = Account.includes(:registrar).search(params[:q])
+      @accounts = @q.result.page(params[:page])
+      @accounts = @accounts.per(params[:results_per_page]) if paginate?
+
+      render_by_format('admin/accounts/index', 'accounts')
+    end
+
+    def show; end
+
+    def edit; end
+
+    def update
+      if @account.valid?
+        @sum = params[:account][:balance].to_f - @account.balance
+        redirect_to admin_accounts_path, notice: t('.updated') if create_activity
+      else
+        render 'edit'
+      end
+    end
+
+    private
+
+    def create_activity
+      activity = AccountActivity.new(account: @account,
+                                     sum: @sum,
+                                     currency: @account.currency,
+                                     description: params[:description],
+                                     activity_type: AccountActivity::UPDATE_CREDIT)
+
+      if activity.save
+        true
+      else
+        false
+      end
+    end
```

Comment:
Small mark, although not essential. Just for the beauty of the code
```
return true if activity.save
false
```

Reply:
Although I could even call like `save!`, so that if it fails to save, it throws an exception about the reasons for the error. Then there is no need for conditions. It seems to me that if it returns false, it will take time trying to debug the code.

Reply:
Well, that could be done like that, but not there.

Considered a good habit to move all stuff not related to action rendering (i.e. business logic) from controllers to the service layer- to service objects/interactors etc. See David Copeland, Sustainable Web Development with Ruby on Rails (https://www.amazon.com/Sustainable-Web-Development-Ruby-Rails-ebook/dp/B08ND6X37Q)